agent/containers/images/image-test-script.sh

```diff
@@ -0,0 +1,283 @@
+wait_keypress 120
+
+# Option to start grafana, where it is listening on port 3000.
+printf -- "\n\nYou can also now run a live metrics visualizer for the Prometheus & PCP data\n\n\t$ podman run --network host -d --rm --name pbench-viz quay.io/pbench/live-metric-visualizer\n\n"
+
+printf -- "\n\nIf done, open a browser to watch live metrics at: %s\n\n" "http://$(hostname -f):3000/"
+wait_keypress 120
+
+
+# TBD start/stop/send tools
+
+printf -- "\n\nNow we will start and stop two iterations of transient tool data collection.\n"
+
+sample="sample42"
+iterations="0-iter-zero 1-iter-one"
+
+> ${benchmark_run_dir}/.iterations
+
+for iteration in ${iterations}; do
+    echo "${iteration}" >> ${benchmark_run_dir}/.iterations
+    benchmark_results_dir="${benchmark_run_dir}/${iteration}/${sample}"
+    mkdir -p ${benchmark_results_dir}
+
+    printf -- "\n\nStarting iteration '${iteration}'; when we continue the transient tools will be started.\n\n"
+    wait_keypress 120
+
+    pbench-start-tools --group="${group}" --dir="${benchmark_results_dir}"
+
+    printf -- "\n\nTransient tools have started for iteration '${iteration}'; when we continue they'll be stopped.\n\n"
+    wait_keypress 120
+
+    pbench-stop-tools --group="${group}" --dir="${benchmark_results_dir}"
+
+    printf -- "\n\nTools have stopped for iteration '${iteration}'; each Tool Meister still has the data held locally.\n\n"
+    wait_keypress 120
+done
+
+printf -- "\n\nWe have completed our two iterations, and next we'll loop through those iterations requesting the tool data be sent back to the Tool Data Sink.\n\n"
+wait_keypress 120
+
+for iteration in ${iterations}; do
+    benchmark_results_dir="${benchmark_run_dir}/${iteration}/${sample}"
+    pbench-send-tools --group="${group}" --dir="${benchmark_results_dir}"
+done
+
+
+# Stop the Tool Meisters, collecting system information, and stopping any persistent tools.
+
+printf -- "\n\nWe have gathered the transient data from our two iterations, and next we'll stop the Tool Meisters; this involves ending any persistent tools, and gathering the final 'sysinfo' collection requested.\n\n"
+wait_keypress 120
+
+set -x
+_PBENCH_TOOL_MEISTER_STOP_LOG_LEVEL=debug pbench-tool-meister-stop --sysinfo=default ${group}
+set +x
+
+
+printf -- "\n\nAt this point the Tool Data Sink has stopped, along with the Tool Meisters. The Redis server is still running, since the pbench-agent CLI commands did not start it.  Next we dump the final directory hierarchy of collected data to complete the test.\n\n"
+wait_keypress 120
+
+# Dump our local environment
+find ${pbench_run} -ls | less -S
+
+wait_keypress 120
+
+read -p "Would you like the live-metric-visualizer to be terminated if started locally? (y/n): " term_vis
```

Comment:
It seems like you could answer the "if started locally" programmatically and then skip this question if it's not there.

Reply:
Ditto for the Redis container at line 276.